### PR TITLE
Remove deprecated pre-commit script and adjust doc

### PR DESCRIPTION
**Pull request by ewjoachim**
Ping @hauntsaninja :)

As suggested in https://github.com/python/typeshed/pull/6341#issuecomment-982470714

### CODE REVIEW

File: CONTRIBUTING.md
Comment (Akuli) [inline CONTRIBUTING.md:36]: Why would anyone want this? It would only run black and isort, and those run automatically when you make a pull request anyway.

File: CONTRIBUTING.md
Comment (Akuli) [inline CONTRIBUTING.md:34]: It would be nicer to say `pre-commit.ci`, to make it clear that you don't have to run anything locally.

File: CONTRIBUTING.md
Comment (ewjoachim) [reply CONTRIBUTING.md:36]: While I understand this might not be everyone's choice, I appreciate that some people like to post clean commits. Also on a more pragmatic point of view, this avoids launching about 40 jobs to cancel them after a few seconds, which has an (albeit small, yet) existing environmental impact :)

But I understand your point, in that we can make it clearer that this step is 100% optional.

Comment (ewjoachim) [reply CONTRIBUTING.md]: Is it good for you now ?

File: CONTRIBUTING.md
Comment (ewjoachim) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md
Comment (AlexWaygood) [inline CONTRIBUTING.md:32]:
```suggestion
Note that this step is entirely optional, as your code will be
automatically reformatted when your PR is submitted
(see **Code Formatting** below),
and the full test suite will also be run automatically.

Follow platform-specific instructions below.
```

File: CONTRIBUTING.md
Comment (Akuli) [inline CONTRIBUTING.md:82]: Nit:
```suggestion
right away and add a commit to your PR. Neat, no?
```

File: CONTRIBUTING.md
Comment (Akuli) [reply CONTRIBUTING.md]: Thanks! I like it now :)

File: CONTRIBUTING.md
Comment (ewjoachim) [reply CONTRIBUTING.md:32]: I'll try to make a commit along those lines :)

Comment (AlexWaygood) [reply CONTRIBUTING.md:32]: I absolutely agree that running the tests locally is valuable! But in the **Contribution process at a glance** section, it says "Optionally run the tests". The wording here sounds much stronger at the moment -- I think the two sections should be consistent with each other 🙂 Also, I've often found typeshed's CI to be _far_ faster than my machine (but, maybe that's just a sign I need a new laptop...)

**Review by Akuli — APPROVED** [review]
Looks good! I'll wait for another maintainer to read this too before merging.

File: CONTRIBUTING.md
Comment (hauntsaninja) [inline CONTRIBUTING.md:33]:
```suggestion
Typeshed runs continuous integration (CI) on all pull requests. This will
automatically fix formatting (using `black`, `isort`) and run tests.
It means you can ignore all local setup on your side, focus on the
code and rely on the CI to fix everything, or point you to the places that
need fixing.
```

**Comment by hauntsaninja** [comment]
Thank you! :-)

**Pull request merged**
